src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,95 @@
+import usersFromServer from '../../api/users';
+import { Todo } from '../../types/Todo';
+import React, { FC, useState, FormEvent } from 'react';
+import { findUserById } from '../../utils/FindUserById';
+
+interface Props {
+  onAdd: (todo: Todo) => void;
+  todos: Todo[];
+}
+
+export const TodoForm: FC<Props> = ({ onAdd, todos }) => {
+  const [newTitle, setNewTitle] = useState('');
+  const [newUserId, setNewUserId] = useState(0);
+
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+
+  function handleFormReset() {
+    setNewTitle('');
+    setNewUserId(0);
+    setTitleError('');
+    setUserError('');
+  }
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    if (!newTitle) {
+      setTitleError('Please enter a title');
+    }
+
+    if (!newUserId) {
+      setUserError('Please choose a user');
+    }
+
+    if (newTitle && newUserId) {
+      const nextId: number = todos[todos.length - 1].id + 1;
```

Comment:
after u remove the sort at the beginning - u can use` Math.max(a, b, c)` for max id here
but note that for an empty array - `Math.max` returns `Infinity` value